### ISSUE TITLE
xml issues

### ISSUE THREAD

**Issue by ingbrzy** [deivids84/MA-XML-7.0-CATALAN#36]
remove all untranslatables!

http://translators.xiaomi.eu/XML_MIUI8-Catalan-ca.html

**Comment by ingbrzy** [comment]
fixed by 3161e2cb1dcbf91ca14d31e2a9cb2403f57dbd94

**Comment by leixet** [comment]
Hi @ingbrzy ! Sorry and thanks for letting us know this error.

Although I think I've read all the available documentation for Xiaomi.eu translators, I haven't found the exact instructions in how to organize the repo: differences between main and devices folder (which *.apk folders must contain each one), also the presence and content of extra folder, and also which exactly strings or xml files have not to be included as aren't translateable.

Can we found a doc (or web or readme file) explaining this?

Now we have downloaded the gemini strings from your v8-XML-Compare repo, and we are translating and correcting this files and putting translated files into main folder in our repo (here), but what we must do with other devices folders? (and the new gemini stable one?)

Sorry for that. We are trying to work in the best way possibe and make a good translation.

**Comment by ingbrzy** [comment]
device folder contain only device specific strings/arrays if there are any... all other translation or xmls should be in main folder to get apply for all devices...

there is one important rule to follow: 
What you dont translate (not translatable), dont put to your translation here.. like any values in arrays/strings which cause functions misbehave or app crash... 

regarding stable xmls, dont bother with that.. stable releases are older than weekly so if you keep weekly updated, they fit stable also..

after your commit always check log here http://translators.xiaomi.eu/XML_MIUI8-Catalan-ca.html
its checking translation every 15min and if you have any issue there, fix it then...

**Comment by leixet** [comment]
Yes, I know about this checking procedure. I've checked yesterday but I forgot that it is not immediate, and didn't wait this 15 minutes to check it again. Sorry. We will do it every time we commit any files from now on.

Regarding device specific or common strings, how can we know which strings are common? We expect to have a "main" folder in your "v8-XML-Compare" repo, but there are only specific device folders. For the moment, we put all files from Gemini in the main folder. How can we know which strings files from other devices folders are specific and must be putted in device specific folder? Then it will be correct having all Gemini strings as common and no specific strings in a separate Gemini folder? Only other devices?

Thank you very much @ingbrzy for your quick answer and sorry for any inconvenience.

**Comment by ingbrzy** [comment]
if you choose gemini as main, then all whats different from gemini in other devices goes to device folders.. we dont have automatic check what suppose to be in device folder so you have to compare gemini vs other devices xmls for differences... for compare folders use Oxygen XML...

**Comment by leixet** [comment]
Perfect! Thank you @ingbrzy! I assume that if I find an apk folder with the same name in two different devices, then they have the same content (exact strings), isn't it?

Also, and finally, there is some exact rule to know which files/strings are not translatable? There are cases that it's very clear, but in others, it's difficult to tell. If it's a whole file I've understood that we have to NOT put it on this repo, but if there are single strings in a file with other translatable strings, we must remove this strings? Whole line? Or only leaving the translated part blank?

**Comment by ingbrzy** [comment]
not exactly.. device folder is for same apk but with different translation specific for that device only... 

common not translatable are arrays and strings with "value" in name... yea just remove that arrays or strings from your translation at all..

**Comment by leixet** [comment]
Ok. I've now commited some files with new translations and see what strings must be deleted as are not translatable. As I see on the XML check log, I think now it's all correct.

The problem is that we are translating the files in Crowdin, and the original strings in Crowdin include the strings that are not translatable, because we have the Crowdin project synced with a repo I've forked from your v8-XML-Compare repo. So when I download the files from Crowdin to commit them here, I have to look inside each file to manually delete the strings that "I think" are no translatable.

As it seems you have an algorithm that finds all the untranslatable strings (the XML checker in translators.xiaomi.eu) it will be useful to have this sort of algorithm or rules to automatically delete all these strings from our files even before uploading it to Crowdin, and saving work to our translators... It's possible that or something similar, @ingbrzy ?
